### DISCUSSION TITLE
Enhancement features

### DISCUSSION THREAD

**Discussion by sachinjung** [amidaware/tacticalrmm#1601] [Ideas]
While showing the demo on RMM side for a small sized MSP I got a quick idea. We can add the following features to enhance the solution.
1. Ticketing and Chat/Call feature: We can add ticketing feature so that customer or agent can create support ticket and assign the ticket to a specific agent. Agent can leverage the chat feature to do direct messages so that support is effective.
2.  Asset management: We do have a great view of Multi-Tenant view as well as host listing but we can enhance this by adding asset management where we can add or register assets like network devices, servers, PCs, Firewall, Printer, IP Phones, IT Racks (DC Racks) and unmanageable devices. If we can add the asset management then we can create SSH, SMB or telnet like connections to those devices for support as well.
3. I have been working in DC Infra and cybersecurity for 18 years so I would like to recommend adding a vulnerability management platform within this solution (May be enterprise feature) which can perform a vulnerability scanning and provide a detailed report. On the basis of that report the support agent can schedule or apply patches or apply configurations that can be fixed. This way we can also leverage the compliance feature as well.
4. Report creation: If we can add report creation on compliance, support hours, activities, and dedicated timestamp then we can add billing as well. So I would like to recommend adding reporting and billing feature as well.

I know the features I have recommended may be in your pipeline or will be added to the pipeline and requires a lot of time so I hope to see these improvements in near future.

**Comment by 3eeKeeper** [comment]
I'd like to add to this, can we integrate a Windows backup? or another backup server like Amanda?

Antivirus integration using their APIs (Malwarebytes endpoint has an API)

**Reply by silversword411** [reply]
AV...has an API?

Don't tell malware makers ;) I'm sure it requires a key....

We have scripts for monitoring Veeam and Urbackup.

Windows/Amanda backup scripts/monitoring would be welcome https://docs.tacticalrmm.com/contributing_community_scripts/

**Comment by silversword411** [comment]
We have a long wishlist, and all those things are major work. I think some are already written up.

In the mean time, there's huge power in the scripting system...and if you can do some scripts in the mean time to integrate those things I'm happy to help out where I can. Goto to Discord and chat about it in #scripts channel

**Comment by ConvexSERV** [comment]
I know this is a few months old, but I thought I would add.

We're new to Tactical (just subscribed today). 
We use Netbox (https://github.com/netbox-community/netbox) for asset management. I'd love to see an integration between the two projects. Both are DJango-based and both have an API.

**Reply by dinger1986** [reply]
thanks for joining us! The biggest things with integrations generally is the mount of time required to put them in place. The best place to start is either someone in the community using both projects (or investigating both projects) and being prepared to spend some time looking at them and getting something going (with assistance from both projects) or a willing ness from the other project to work together to build out an integration benefitting both communities. It would be hard, albeit not impossible for us to integrate alot of projects but we want to work with projects who want to work with us.

**Reply by ConvexSERV** [reply]
Totally understand.
We're pretty invested in Netbox for inventory, and I've started re-working our custom scripts from our previous RMM (Datto) so we can cut the Kaseya cord. I'll upload the scripts to the community once I'm confident we have them stable.

I am certainly willing to work on integration features. I'll see what I can put together and will report back when I have something that might be worth sharing.  Is this the best forum for reporting back? 

Thanks.

**Reply by dinger1986** [reply]
Sounds perfect.

Discord is probably best if you want some others jumping in and helping with things I'm happy to jump in on some scripts etc

**Reply by silversword411** [reply]
an n8n recipe that will do the integration syncing IMO is the best place to put those efforts.
